x-pack/legacy/plugins/siem/server/lib/timeline/routes/import_timelines_route.test.ts

```diff
@@ -0,0 +1,339 @@
+  let securitySetup: SecurityPluginSetup;
+  let { context } = requestContextMock.createTools();
+  let mockGetTimeline: jest.Mock;
+  let mockPersistTimeline: jest.Mock;
+  let mockPersistPinnedEventOnTimeline: jest.Mock;
+  let mockPersistNote: jest.Mock;
+
+  beforeEach(() => {
+    jest.resetModules();
+    jest.resetAllMocks();
+    jest.restoreAllMocks();
+    jest.clearAllMocks();
+
+    server = serverMock.create();
+    context = requestContextMock.createTools().context;
+    config = jest.fn().mockImplementation(() => {
+      return mockConfig;
+    });
+
+    securitySetup = ({
+      authc: {
+        getCurrentUser: jest.fn().mockReturnValue(mockGetCurrentUser),
+      },
+      authz: {},
+    } as unknown) as SecurityPluginSetup;
+
+    mockGetTimeline = jest.fn();
+    mockPersistTimeline = jest.fn();
+    mockPersistPinnedEventOnTimeline = jest.fn();
+    mockPersistNote = jest.fn();
+
+    jest.doMock('../create_timelines_stream_from_ndjson', () => {
+      return {
+        createTimelinesStreamFromNdJson: jest.fn().mockReturnValue(mockParsedObjects),
+      };
+    });
+
+    jest.doMock('../../../../../../../../src/legacy/utils', () => {
+      return {
+        createPromiseFromStreams: jest.fn().mockReturnValue(mockParsedObjects),
+      };
+    });
+
+    jest.doMock('./utils/import_timelines', () => {
+      const originalModule = jest.requireActual('./utils/import_timelines');
+      return {
+        ...originalModule,
+        getTupleDuplicateErrorsAndUniqueTimeline: jest
+          .fn()
+          .mockReturnValue([mockDuplicateIdErrors, mockUniqueParsedObjects]),
+      };
+    });
+  });
+
+  describe('Import a new timeline', () => {
+    beforeEach(() => {
+      jest.doMock('../saved_object', () => {
+        return {
+          Timeline: jest.fn().mockImplementation(() => {
+            return {
+              getTimeline: mockGetTimeline.mockReturnValue(null),
+              persistTimeline: mockPersistTimeline.mockReturnValue({
+                timeline: { savedObjectId: '79deb4c0-6bc1-11ea-9999-f5341fb7a189' },
+              }),
+            };
+          }),
+        };
+      });
+
+      jest.doMock('../../pinned_event/saved_object', () => {
+        return {
+          PinnedEvent: jest.fn().mockImplementation(() => {
+            return {
+              persistPinnedEventOnTimeline: mockPersistPinnedEventOnTimeline,
+            };
+          }),
+        };
+      });
+
+      jest.doMock('../../note/saved_object', () => {
+        return {
+          Note: jest.fn().mockImplementation(() => {
+            return {
+              persistNote: mockPersistNote,
+            };
+          }),
+        };
+      });
+
+      const importTimelinesRoute = jest.requireActual('./import_timelines_route')
+        .importTimelinesRoute;
+      importTimelinesRoute(server.router, config, securitySetup);
+    });
+
+    test('should see if the given timeline savedObjectId already exist', async () => {
```

Comment:
consider revising this description to make it clearer how the assertion tests this case

Reply:
Agree, I still have more cases to do!